core/src/main/java/cucumber/runtime/Stats.java

```diff
@@ -94,7 +94,7 @@ public void setEventPublisher(EventPublisher publisher) {
 
     public byte exitStatus(boolean isStrict) {
         byte result = 0x0;
-        if (!failedScenarios.isEmpty() || (isStrict && (!pendingScenarios.isEmpty() || !undefinedScenarios.isEmpty()))) {
+        if (!failedScenarios.isEmpty() || (isStrict && (!pendingScenarios.isEmpty() || !undefinedScenarios.isEmpty() || !ambiguousScenarios.isEmpty()))) {
```

Comment:
Looking at [printNonZeroResultScenarios](https://github.com/cucumber/cucumber-jvm/pull/1342/files#diff-9c2af73a8f4e34a02a035f2709e487d5R157) I think `ambiguousScenarios` should be treated the same as `failedScenarios` and always return a non zero exit status.